core/domain/opportunity_services.py

```diff
@@ -355,8 +368,16 @@ def update_exploration_opportunities_with_story_changes(
         exploration_opportunity_summary_list.append(
             exploration_opportunity_summary)
 
-    _save_multi_exploration_opportunity_summary(
-        exploration_opportunity_summary_list)
+    exploration_opportunity_summary_model_list = (
```

Comment:
The old method is repeated multiple times below -- I think you should keep it and have it do the update-and-put. Less duplication.